ext/spl/spl_directory.c

```diff
@@ -599,84 +598,124 @@ static inline HashTable *spl_filesystem_object_get_debug_info(zend_object *objec
 {
 	spl_filesystem_object *intern = spl_filesystem_from_obj(object);
 	zval tmp;
-	HashTable *rv;
-	zend_string *pnstr;
-	zend_string *path;
-	char stmp[2];
+	HashTable *debug_info;
+	zend_string *mangled_named;
+	zend_string *path_name;
 
 	if (!intern->std.properties) {
 		rebuild_object_properties(&intern->std);
 	}
 
-	rv = zend_array_dup(intern->std.properties);
+	// TODO Do zend_new_array() + zend_hash_copy() trick?
+	debug_info = zend_array_dup(intern->std.properties);
 
-	pnstr = spl_gen_private_prop_name(spl_ce_SplFileInfo, "pathName", sizeof("pathName")-1);
-	path = spl_filesystem_object_get_pathname(intern);
-	if (path) {
-		ZVAL_STR_COPY(&tmp, path);
+	mangled_named = zend_mangle_property_name(
+		ZSTR_VAL(spl_ce_SplFileInfo->name),
+		ZSTR_LEN(spl_ce_SplFileInfo->name),
+		"pathName",
+		strlen("pathName"),
+		/* persistent */ false
+	);
+	path_name = spl_filesystem_object_get_pathname(intern);
+	if (path_name) {
+		ZVAL_STR_COPY(&tmp, path_name);
 	} else {
 		ZVAL_EMPTY_STRING(&tmp);
 	}
-	zend_symtable_update(rv, pnstr, &tmp);
-	zend_string_release_ex(pnstr, 0);
+	zend_hash_update(debug_info, mangled_named, &tmp);
+	zend_string_release_ex(mangled_named, /* persistent */ false);
+	/* IMPORTANT: Do not free path_name */
```

Comment:
If you write comments like this, include the reason why

Reply:
I haven't figured out the reason why yet, I'll investigate more.